### PR TITLE
Update request logger to support v2 protocol

### LINKED ISSUE #2791: Handle KFServing V2 Protocol in request logger
* Update request logger and tests
 * Update feedback in executor to pass directly to request logger

### PR DESCRIPTION
**Pull request by cliveseldon**
**What this PR does / why we need it**:

  * Support the V2 protocol in the request logger

**Which issue(s) this PR fixes**:

Fixes #2791 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **cliveseldon**
You can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Wed Dec 23 14:50:22 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2802/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2802 --build=2

**Comment by seldondev** [comment]
Wed Dec 23 14:50:43 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2802/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2802 --build=1

File: components/seldon-request-logger/app/default_logger.py
Comment (ryandawsonuk) [inline components/seldon-request-logger/app/default_logger.py:276]: Just comparing this logic to https://github.com/SeldonIO/seldon-core/blob/master/components/seldon-request-logger/app/default_logger.py#L314
In that logic 3 dimensions would be batch tabular. Has me wondering about the assumption there that the first dimension is batch.

**Comment by seldondev** [comment]
Tue Jan  5 15:57:15 UTC 2021
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2802/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2802 --build=3

**Comment by seldondev** [comment]
Tue Jan  5 15:57:24 UTC 2021
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2802/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2802 --build=4

**Comment by cliveseldon** [comment]
BYTES data fails. But we could leave this until we have a use case.

**Review by ryandawsonuk — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
